src/app/features/test-records/views/test-record/test-record.component.ts

```diff
@@ -48,7 +50,7 @@ export class TestRecordComponent implements OnInit, OnDestroy {
     this.testResult$ = this.testRecordsService.editingTestResult$.pipe(
       switchMap(editingTestResult => (editingTestResult ? of(editingTestResult) : this.testRecordsService.testResult$))
     );
-    this.defects$ = this.testRecordsService.defectData$;
+    this.defectsStore.dispatch(fetchDefects());
```

Comment:
Could we abstract this to a service ? and same for line 69

Reply:
No

Reply:
(The explanation for this was given in our discussion 🙂)